packages/bitcore-node/src/models/transaction.ts

```diff
@@ -673,29 +673,33 @@ export class TransactionModel extends BaseTransaction<IBtcTransaction> {
   async _invalidateTx(params: {
     chain: string;
     network: string;
-    invalidTxid: string,
-    replacedByTxid?: string // only provided at the beginning of the ancestral tree. Txs that spend unconfirmed outputs aren't "replaced"
-    invalidParentTxids?: string[] // empty at the beginning of the ancestral tree. 
+    invalidTxid: string;
+    replacedByTxid?: string; // only provided at the beginning of the ancestral tree. Txs that spend unconfirmed outputs aren't "replaced"
+    invalidParentTxids?: string[]; // empty at the beginning of the ancestral tree.
+    simple?: boolean; // if true, don't invalidate descendants
   }) {
-    const { chain, network, invalidTxid, replacedByTxid, invalidParentTxids = [] } = params;
-    const spentOutputsQuery = {
-      chain,
-      network,
-      spentHeight: SpentHeightIndicators.pending,
-      mintTxid: invalidTxid
-    };
+    const { chain, network, invalidTxid, replacedByTxid, invalidParentTxids = [], simple } = params;
+    
+    if (!simple) {
+      const spentOutputsQuery = {
+        chain,
+        network,
+        spentHeight: SpentHeightIndicators.pending,
+        mintTxid: invalidTxid
+      };
 
-    // spent outputs of invalid tx
-    const spentOutputsStream = await CoinStorage.collection.find(spentOutputsQuery);
-    const seenTxids = new Set();
-    let output: ICoin | null;
+      // spent outputs of invalid tx
+      const spentOutputsStream = await CoinStorage.collection.find(spentOutputsQuery);
```

Comment:
Getting rid of the await here could stop someone from making a mistake when working with the Cursor.